ASP/HTML-ASP.sublime-syntax

```diff
@@ -14,9 +14,10 @@ contexts:
         - match: "%>"
           scope: punctuation.section.embedded.end.asp
           pop: true
-        - match: (').*?(?=%>)
-          scope: comment.line.apostrophe.asp
-          captures:
-            1: punctuation.definition.comment.asp
-        - include: scope:source.asp
+        - match: ''
+          push:
+            - include: scope:source.asp # don't directly push, so that the scope doesn't also contain 'source.asp ' as well as our 'source.asp.embedded.html '
+          with_prototype:
+            - match: "(?=%>)"
+              pop: true
```

Comment:
I was thinking about this the other day – should we allow the ASP syntax to embed HTML inside of it instead of popping out of all nested context back up to the parent HTML context?

If we allow embedding HTML inside of nesting ASP contexts, then the ASP syntax should better be able to track when it is still inside of a function, etc. We would probably only want to push into a nesting HTML context if we have entered a nested ASP context. If we are still at the top-most level, we should probably still use the match of `%>` in the initial `<%=` push context.

Does the question make sense, or should I try to explain my idea a different way?